### ISSUE TITLE
new-cap's capIsNewExceptions false warning

### ISSUE THREAD

**Issue by sebastienbarre** [eslint/eslint#2023]
I'm using 0.16.2. This code:
```
'use strict';
var Immutable = require('immutable');
var foo = Immutable.List();
```
triggers a ```new-cap``` warning for me, ```A function with a name starting with an uppercase letter should only be used as a constructor```, even though I have an exception set:
```
    "new-cap": [2, {
      "capIsNewExceptions": ["Immutable"]
    }]
```
Thank you
For reference: http://facebook.github.io/immutable-js/

**Comment by ilyavolodin** [comment]
Can you try changing your configuration to this:
```json
"new-cap": [2, { "capIsNewExceptions": ["List"] }]
```

Looking through the code of the rule, it looks like it's actually complaining about List, not Immutable.

**Comment by gyandeeps** [comment]
I think you are right @ilyavolodin as I have experienced same issue before.

**Comment by sebastienbarre** [comment]
You are correct, this is indeed the issue. It threw me off, as ```Immutable``` is the word being selected (```9 characters selected```) and highlighted. 

Hard to make case for putting ```"List"``` in ```capIsNewExceptions``` though. Who knows if an actual ```List``` class might require ```new``` down the road...

Another workaround would be:
```
'use strict';
var Immutable = require('immutable');
var ImmutableList = Immutable.List;
var foo = ImmutableList();
```
and
```
    "new-cap": [2, {
      "capIsNewExceptions": ["ImmutableList"]
    }]
```
but that Immutable library is fairly big, creating a new var for each member could get old.

Thanks for you help guys.

**Comment by sebastienbarre** [comment]
Update:
```
var foo = new Immutable.List();
```

seems to work. Let me check with the author, if that's an anti-pattern.

**Comment by ilyavolodin** [comment]
@sebastienbarre Would you mind opening a new issue for incorrect position in the error message. That should be fixed.

**Comment by sebastienbarre** [comment]
@ilyavolodin Done in #2025

**Comment by leebyron** [comment]
It is a factory function, not a constructor, much like ES6's Symbol() function. 

You can call it with new if you like, but the new'd memory is not used, so you're just unnecessarily churning GC.

IMHO, the new-cap rule is a dangerous one. While it can sometimes catch missing "new" keywords, there are just too many edge-cases where inserting "new" is erroneous. I've always turned this rule off for my own projects, ever since jslint.

**Comment by sebastienbarre** [comment]
Thanks @leebyron ; I just switched from ```jshint``` to ```eslint```. Was on the fence about ```new-cap```, thanks for your input, better left off at the moment. Immutable rocks BTW.

**Comment by nzakas** [comment]
It seems like #2025 is the real issue here, so closing.

**Comment by yvele** [comment]
This exception is working for Immutable:

```json
"new-cap": [2, { "capIsNewExceptionPattern" : "^Immutable\\.." }]
```

PS: About immutable new-cap convention: https://github.com/facebook/immutable-js/issues/10
